BinaryEncoding.md

```diff
@@ -1,122 +1,400 @@
-their contents and children using dense integer indices and without any kind of embedded metadata 
-or tagging. This raises the question of how to reconcile the efficient encoding with the 
-backwards-compatibility goals.
-
-Specifically, we'd like to avoid the situation where a future version of WebAssembly has features 
-F1 and F2 and vendor V1 implements F1, assigning the next logical operator indices to F1's new
-operators, and V2 implements F2, assigning the same next logical operator indices to F2's new operators 
-and now a single binary has ambiguous semantics if it tries to use either F1 or F2. This type of 
-non-linear feature addition is commonplace in JavaScript and Web APIs and is guarded against by 
-having unique names for unique features (and associated [conventions](https://hsivonen.fi/vendor-prefixes/)).
-
-The current proposal is to maintain both the efficiency of indices in the [serialized AST](BinaryEncoding.md#serialized-ast) and the established
-conflict-avoidance practices surrounding string names:
-  * The WebAssembly spec doesn't define any global index spaces
-    * So, as a general rule, no magic numbers in the spec (other than the literal [magic number](https://en.wikipedia.org/wiki/Magic_number_%28programming%29)).
-  * Instead, a module defines its *own* local index spaces of operators by providing tables *of names*. 
-    * So what the spec *would* define is a set of names and their associated semantics.
-  * To avoid (over time) large index-space declaration sections that are largely the same
-    between modules, finalized versions of standards would define named baseline index spaces
-    that modules could optionally use as a starting point to further refine.
-    * For example, to use all of [the MVP](MVP.md) plus
-      [SIMD](PostMVP.md#fixed-width-simd) the declaration could be "base"
-      followed by the list of SIMD operators used.
-    * This feature would also be most useful for people handwriting the [text format](TextFormat.md).
-    * However, such a version declaration does not establish a global "version" for the module
-      or affect anything outside of the initialization of the index spaces; decoders would
-      remain versionless and simply add cases for new *names* (as with current JavaScript parsers).
-
-## Proposals
-
-The native prototype built for [V8](https://github.com/WebAssembly/v8-native-prototype)
-implements a binary format that embodies most, but not all of the ideas in this document.
-It is described in detail in a [public design doc](https://docs.google.com/a/google.com/document/d/1761v1AfhFM5kE8NArF_PyXcl-iVh0Dx3InOrmcyIoiI/pub) and a [copy of the original](https://docs.google.com/document/d/1-G11CnMA0My20KI9D7dBR6ZCPOBCRD0oCH6SHCPFGx0/edit?usp=sharing).
+
+  * Given a call AST node: `struct Call { uint32_t callee_index; vector<AstNode*> args; }`
+    * First write the binary sequence of `Call` (uint8)
```

Comment:
'binary sequence of `Call`' is not very meaningful. The word 'opcode' seems appropriate, and the data type uint8.